topostats/tracing/dnatracing.py

```diff
@@ -314,101 +315,65 @@ def get_splined_traces(self):
         This function actually calculates the average of several splines which is important for getting a good fit on
         the lower res data"""
 
-        step_size = int(7e-9 / (self.pixel_to_nm_scaling))  # 3 nm step size
-        interp_step = int(1e-10 / self.pixel_to_nm_scaling)
-        # Lets see if we just got with the pixel_to_nm_scaling
-        # step_size = self.pixel_to_nm_scaling
-        # interp_step = self.pixel_to_nm_scaling
-
-        self.splining_success = True
-        nbr = len(self.fitted_trace[:, 0])
-
-        # Hard to believe but some traces have less than 4 coordinates in total
-        if len(self.fitted_trace[:, 1]) < 4:
-            self.splined_trace = self.fitted_trace
-            # continue
-
-        # The degree of spline fit used is 3 so there cannot be less than 3 points in the splined trace
-        while nbr / step_size < 4:
-            if step_size <= 1:
-                step_size = 1
+        # Fitted traces are Nx2 numpy arrays of coordinates
+        # All self references are here for easy turning into static method if wanted, also type hints and short documentation
+        fitted_trace = self.fitted_trace  # : np.ndarray - boolean 2d numpy array of fitted traces to spline
+        step_size_m = self.step_size_m  # : float - the step size for the splines to skip pixels in the fitted trace
+        pixel_to_nm_scaling = self.pixel_to_nm_scaling  # : float - pixel to nanometre scaling factor for the image
+        mol_is_circular = self.mol_is_circular  # : mol_is_circular - whether or not the molecule is classed as circular
+        n_grain = self.n_grain  # : int - the grain index (for logging purposes)
+
+        # Calculate the step size in pixels
+        step_size_px = int(step_size_m / pixel_to_nm_scaling)
+
+        # Splines will be totalled and then divived by number of splines to calculate the average spline
+        spline_sum = None
+        nbr = len(fitted_trace)
+
+        fitted_trace_length = fitted_trace.shape[0]
+
+        if fitted_trace_length < 4:
```

Comment:
Why 4? Wouldn't this depend on the step size in pixels?:
```suggestion
        if fitted_trace_length < step_size_px:
```

Reply:
This is the minimum number of points it's okay to ask it to spline for the hard-coded degree polynomial of 3. Since I've now made the polynomial degree a configurable parameter, I've updated this accordingly. 👍